butterfree/transform/transformations/transform_component.py

```diff
@@ -1,10 +1,13 @@
 """Transform Abstract Class."""
 
 from abc import ABC, abstractmethod
-from typing import Any, List
+from typing import TYPE_CHECKING, List, Optional, Tuple
 
 from pyspark.sql import DataFrame
 
+if TYPE_CHECKING:
```

Comment:
in what context would `TYPE_CHECKING` not be used?

Reply:
TYPE_CHECKING is a constant from the typing module to control imports only for type checking, without impacting the code execution time